components/tools/OmeroJava/test/integration/AbstractServerTest.java

```diff
@@ -285,6 +298,153 @@ public void tearDown() throws Exception {
     }
 
     /**
+     * Cast the map of strings (e.g. {@link #ALL_GROUPS_CONTEXT})
+     * into implicit context, which asks for {@code <String, String>}.
+     * @param implicitContext the implicit context to be changed
+     * @param newContext the map of strings (e.g. {@link #ALL_GROUPS_CONTEXT})
+     */
+    protected void mergeIntoContext(ImplicitContext implicitContext, Map<String, String> newContext) {
+        for (final Entry<String, String> entry : newContext.entrySet()) {
+            implicitContext.put(entry.getKey(), entry.getValue());
+        }
+    }
+
+    /**
+     * Add the given annotation to the given image.
+     * @param image an image
+     * @param annotation an annotation
+     * @return the new loaded link from the image to the annotation
+     * @throws ServerError an error possibly occurring during saving of the link
+     */
+    protected ImageAnnotationLink linkImageAnnotation(Image image, Annotation annotation) throws ServerError {
+        if (image.isLoaded() && image.getId() != null) {
```

Comment:
Could image or annotation be ``null``?  I guess the test will be wrong if this is the case